Assets/Scripts/Controllers/InputOutput/SoundController.cs

```diff
@@ -28,6 +28,29 @@ public void Update(float deltaTime)
         soundCooldown -= deltaTime;
     }
 
+	//This is a delegate to play "PlayClick" on ShowDialog and CloseDialog from DialogBox script. 
+	delegate void ClickDelegate();
+		ClickDelegate btnsfx;
+	public void PlayClick()
```

Comment:
the purpose of a delegate is to be used by another class without the original class with the delegate needing to know the other class exists. Because this delegate is never referenced by another file it can be deleted and you can just call OnButtonClickedSFX directly inside of PlayClick.

ideally you would create a delegate (or System.Action, their purposes are identical but Action is a bit easier to use) in the dialogBox class and then add OnButtonClickedSFX to it on line 21 of this file, kind of like the rest of the sounds in this file. However, this approach would be very difficult because DialogBox is never created directly, its only a parent to other specific dialog boxes, so you don't have to worry about it in this PR.